src/app/plugins/scores.plugin.ts

```diff
@@ -62,81 +72,155 @@ export class ScoresPlugin extends Plugin {
-        if (embedBuffer.length >= this._MAX_NUM_DISPLAY) { break; } //Stop if the max amount of messages have been queued
-
-        teamFound = true; //Designates as true if any game was found with matching search term
-
-        let visitorData = game.match(visitorDataRegex);
-        let homeData = game.match(homeDataRegex);
+        let visitorData = game.match(visitorDataActiveRegex);
+        let homeData = game.match(homeDataActiveRegex);
 
         //Makes sure regex found data
-        if (visitorData == null || homeData == null) { continue; }
-
-        visitorData = visitorData[0];
-        homeData = homeData[0];
+        if (!visitorData || !homeData) {
+          //If this didn't work, it may be an upcoming game
+
+          //Call function and make sure teamFound is true if this is true atleast once
+          if (this._tryUpcomingGame(embedBuffer, game, teamArg)) {
+            teamFound = true;
+          }
+          continue;
+        }
 
-        visitorData = visitorData.split('%20');
-        homeData = homeData.split('%20');
+        visitorData = visitorData[0].split('%20');
+        homeData = homeData[0].split('%20');
 
         visitorData[0] = visitorData[0].slice(1); //Remove the prefix that starts the visitors name
 
         const visitorName = this._getTeamNameFromTeamData(visitorData);
         const homeName = this._getTeamNameFromTeamData(homeData);
 
+        //Check if game does not contain all search terms
+        if (
+          !(
+            this._strictlyContainsAllTokens(visitorName, teamArg) ||
+            this._strictlyContainsAllTokens(homeName, teamArg)
+          )
+        ) {
+          continue;
+        }
+
+        teamFound = true; //Designates as true if any game was found with matching search term
+
         const visitorScore = parseInt(visitorData[visitorData.length - 1]);
         const homeScore = parseInt(homeData[homeData.length - 1]);
 
-        //Boolean for if the searched team is the one winning
-        let winning = 0;
-        let opponentName = '';
-        let teamName = '';
-
         //If searched team is the visitor, else
-        if (this._containsAllTokens(visitorName, teamArg.split(' '))) {
-
-          if (visitorScore > homeScore) { winning = 1; }  //Winning
-          else if (visitorScore < homeScore) { winning = 0; } //Losing
-          else if (visitorScore == homeScore) { winning = 2; } //Tied
-
-          opponentName = homeName;
-          teamName = visitorName;
-        } else {
-
-          if (homeScore > visitorScore) { winning = 1; }  //Winning
-          else if (homeScore < visitorScore) { winning = 0; } //Losing
-          else if (homeScore == visitorScore) { winning = 2; } //Tied
-
-          opponentName = visitorName;
-          teamName = homeName;
-        }
+        const winning = this._strictlyContainsAllTokens(visitorName, teamArg) ?
+          this._evaluateScores(visitorScore, homeScore) : this._evaluateScores(homeScore, visitorScore);
```

Comment:
nit: ugly

Reply:
If i replace the condition with a variable storing it, would that be better? or should I try another method than swapping the parameters in _evalScores?

Reply:
if (cond) {swap}
evaluateScores...

Better even, if you returned -1,0,1 you could just negate it